pkg/auth/manager/serviceaccounts/serviceaccounts.go

```diff
@@ -0,0 +1,88 @@
+package serviceaccounts
+
+import (
+	"context"
+
+	authpb "github.com/cs3org/go-cs3apis/cs3/auth/provider/v1beta1"
+	userpb "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+
+	"github.com/cs3org/reva/v2/pkg/auth"
+	"github.com/cs3org/reva/v2/pkg/auth/manager/registry"
+	"github.com/cs3org/reva/v2/pkg/auth/scope"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+)
+
+type conf struct {
+	ServiceUsers []serviceuser `mapstructure:"service_accounts"`
+}
+
+type serviceuser struct {
+	ID     string `mapstructure:"id"`
+	Secret string `mapstructure:"secret"`
+}
+
+type manager struct {
+	authenticate func(userID, secret string) error
+}
+
+func init() {
+	registry.Register("serviceaccounts", New)
+}
+
+// Configure parses the map conf
+func (m *manager) Configure(config map[string]interface{}) error {
+	c := &conf{}
+	if err := mapstructure.Decode(config, c); err != nil {
+		return errors.Wrap(err, "error decoding conf")
+	}
+	// only inmem authenticator for now
+	a := &inmemAuthenticator{make(map[string]string)}
+	for _, s := range c.ServiceUsers {
+		// TODO: hash secrets
```

Comment:
If we want to hash the password. The password hashes should already be supplied via config.  Hashing it here wouldn't have much benefit.

Reply:
I was thinking about memory leaks or something like this. Prolly not needed here. Removed the comment

Reply:
Did you? :thinking:

Reply:
Yes? https://github.com/cs3org/reva/pull/3926/commits/662ac1dd0061068ce8228211b9ecea7676b55d6f#diff-a94eec5ee3afc75706a275742376a4b8d296b2c9624afeaa72ef5d81f4205111L83

Reply:
There was another one. Sorry. Now It's gone :)

Reply:
Ah, didn't notice the other one ;-)